### ISSUE TITLE
Reqirements for V2.0

### ISSUE THREAD

**Issue by steand** [steand/optolink#5]
With one year experience I like to start a 2.0 after summer. 
In this issues i will discuss new requirements and features for this.

My first idea is to make the handling between OH2 and optolink more flexible. 

1. Change the northbound API to REST (I have no experience in REST; need help for this).
2. More easy generic data model for OH2 -> needs maybe a re implementation of OH2 binding.
2.1 The Idea is, that we implement only a hand full channel-types/data-points  like: Temperature Sensor/Aktor, Pump, valve,ect.
2.2 on the OH2 site a binding that generate the thinks and channels from the discovery (here I must first understand/clarify what OH2 supports) 
3. ....

(German comments are welcome also/ Auch Kommentare in Deutsch sind willkommen)

**Comment by FulvioSpelta** [comment]
Available for testing and for analyze REST interface.
Thanks for the effort.

_My current status_
You know currently i'm using your northbound optolink in order to connect to the vitotronic and a self build minimal script (nodejs) in order to get values from optolink and update, using the rest interface, openhab 1 items.

I'm moving to a new OH2 based controller and, currently, tested ok the same architecture.
I'll try the native OH2 binding in the near future (I need to understand how it works because I'm new in OH2). I've done a quick test installing the binding and discovering successfully the optolink adapter but no more. (Any hints will be welcome :-) )

**Comment by Sbried** [comment]
I like the idea to start an 2.0 version with a "more modern" communication. I was also thinking about MQTT.

I was also thinking about kind of "auto-configuration" of things & channels in OH2. But I don´t know if it is possible to add things and channel apart from what is defined in thing-types.xml in OH2.

I´m open to contribute ...

**Comment by FulvioSpelta** [comment]
Just an open thought: i'd prefer to avoid mqtt because it needs a broker so an additional architectural module to add/install/update etc. My idea is to have a "self contained" object.

**Comment by steand** [comment]
@FulvioSpelta 
Yes I remember. But my goal is to implement a native REST in java.
@Sbried 
auto-configuration" of things & channels: My feeling is that it is possible, my understanding is that the homematic binding has only the bridge defined. I didn't understand concept of device really but If I had more time i going to a deep dive. Maybe we can ask @gerrieg .

**Comment by steand** [comment]
Things can be generic :sunglasses:
see: [Documentation of eclipse smarthome](http://www.eclipse.org/smarthome/documentation/development/bindings/faq.html)
The consequence is that the description of things/channels are moved to optolink.xml. 
Is this what we want? 

First idea's:
OH: We have only one Bridge -> HeadingSystem.
OH: Things and channels are create by discovery.  
optolink: channel value type (like: switch, number,..) has been define in the optolink.xml.
The open point: What we do if optolink.xml is changed? 

- ignore (I think: is no possible)
- delete address  in optolink.xml -> delete channels in OH 
- same for things
- channel renamed -> create new channel in OH && delete or keep old channel?

Final: 
This is going to a generic binding in OH and is not depend to Viessmann.
It looks like more a protocol Interface (like MQTT) extend by a data model (discovery).
Or a implementation of [Web Thing](https://www.w3.org/Submission/wot-model/)
I think, first we need a discussion in the [Community]( https://community.openhab.org/) (I going to start after my trip on "Jakobsweg", middle may)

**Comment by gerrieg** [comment]
I'm loading all metadata from the Homematic gateway at bridge startup in model classes (HmDevice, HmChannel, HmDatapoint). Then i'm generating all ThingTypes, ChannelGroups, ChannelTypes, ... from that data, see [HomematicTypeGeneratorImpl](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/type/HomematicTypeGeneratorImpl.java) 
It's fully dynamic, because defining the XML's for all the different devices would be a huge work.

**Comment by FulvioSpelta** [comment]
> @FulvioSpelta
Yes I remember. But my goal is to implement a native REST in java.

I only like to remember my current usage. I understand and like your goal, I'll be available for contributing (if my knowledge will be sufficient) and for testing.

**Comment by steand** [comment]
I have open the discussion on OH Community [CoAP Binding? yes or no](https://community.openhab.org/t/coap-binding-yes-or-no/25468)
@FulvioSpelta 
by the way: I will think about REST also (parallel to CoAP). But for a OH Implementation CoAP sounds good.

**Comment by c4rd0n** [comment]
hi Stefan,

for your optolink V2.0, I purpose you to add multiples connexions support. Today with the optolink RC1.0, when OH2 is connected, We can't open an other session to otpolink. And when OH2 restarts, optolink keeps the old session open and don't respond to the new instance of OH2.

To solve this problems I replaced the open function by a thread creation.
If you want I can create a pull request : [my branch](https://github.com/c4rd0n/optolink/tree/feature/multithreading).

Thank's for your work,

Best Regards,

Félix

**Comment by steand** [comment]
Thanks, 

I am in France for long time. If I am back I going to test it, middle May. 

Regard Stefan

Am 10. April 2017 22:10:53 MESZ schrieb "Félix CARDON" <notifications@github.com>:
